python/mxnet/gluon/contrib/rnn/rnn_cell.py

```diff
@@ -181,3 +181,126 @@ def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=N
         outputs, _, _, _ = _format_sequence(length, outputs, layout, merge_outputs)
 
         return outputs, states
+
+
+class LSTMPCell(HybridRecurrentCell):
```

Comment:
Can this be implemented as a ModifierCell? The implementation should be similar as ZoneOut.

Reply:
Good idea

Reply:
Is there an easy way to modify state in Modifier cell (especially during unroll)??

Reply:
I don't it's a good abstraction as a modifier cell. The projection only happens on the hidden state of LSTM, not on the cell state. I am not sure what's the expected behavior for GRU cell and RNN cell where the number of states is just one. Also the paper call this LSTMP, it's specific to LSTM. I think it's find to just call it `contrib.LSTMP` which inherits `HybridRecurrentCell `.